### PR TITLE
DRAFT: BEHRT

### PR DESCRIPTION
**Pull request by KennethEnevoldsen**
- Implemented:
  - [x] BEHRTEmbedder (outline is done, ~~missing tests + a refactor to vectors~~) (missing masking)
  - [x] Trainer
    - [x] Add tests for trainer
    - [x] Add checkpointing
    - [x]  Add test for inference
  - [x] BEHRTMLMTask
  - [x] PatientDataset

Q: UNK: The model will never see UNK during training. How to handle? 
A: The BEHRT model has similar issue, we leave it as it for now.

### CODE REVIEW

**Comment by MartinBernstorff** [comment]
See #213 for the final data-structure code. Looking forward to this!

File: psycop/common/sequence_models/loggers/base.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+class Logger(Protocol):
+    run_name: str
+    metrics: list[dict[str, float]]
+
+    def __init__(self, project_name: str, run_name: str):
+        ...
+
+    def log_metrics(self, metrics: dict[str, float]) -> None:
+        ...
```
[the remark this answers is not in the document]
Comment (KennethEnevoldsen) [reply psycop/common/sequence_models/loggers/base.py:12]: fixed

File: psycop/common/sequence_models/loggers/base.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+from typing import Protocol
+
+
+class Logger(Protocol):
+    run_name: str
+    metrics: list[dict[str, float]]
+
+    def __init__(self, project_name: str, run_name: str):
```
[the remark this answers is not in the document]
Comment (KennethEnevoldsen) [reply psycop/common/sequence_models/loggers/base.py:8]: fixed

File: psycop/common/sequence_models/loggers/base.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
+from typing import Protocol
+
+
+class Logger(Protocol):
+    run_name: str
+    metrics: list[dict[str, float]]
```
[the remark this answers is not in the document]
Comment (KennethEnevoldsen) [reply psycop/common/sequence_models/loggers/base.py:6]: fixed

[the remark this answers is not in the document]
Comment (KennethEnevoldsen) [reply psycop/common/sequence_models/loggers/base.py:6]: I have removed this. It is still there for testing, but I don't think it should be a requirement in general

**Pull request merged**